src/library/module.cpp

```diff
@@ -49,17 +49,29 @@ namespace lean {
 struct olean_header {
     // 5 bytes: magic number
     char marker[5] = {'o', 'l', 'e', 'a', 'n'};
-    // 1 byte: version, currently always `1`
-    uint8_t version = 1;
-    // 42 bytes: build githash, padded with `\0` to the right
-    char githash[42];
+    // 1 byte: version, incremented on structural changes to header
+    uint8_t version = 2;
+    // 1 byte of flags:
+    // - whether persisted bignums use GMP or Lean-native encoding
+    uint8_t flags =
+#ifdef LEAN_USE_GMP
+        0b1;
+#else
+        0b0;
+#endif
+    // 33 bytes: Lean version string, padded with '\0' to the right
+    // e.g. "4.12.0-nightly-2024-10-18"
```

Comment:
```suggestion
    // 33 bytes: Lean version string, padded with '\0' to the right
    // e.g. "4.12.0-nightly-2024-10-18". May not be null terminated.
```
according to the docs for strncopy.